MDANSE/Src/MDANSE/Framework/Configurators/IConfigurator.py

```diff
@@ -145,6 +159,18 @@ def __init__(self, name, **kwargs):
 
         self._original_input = ""
 
+    def __str__(self) -> str:
+        return "\n".join(
+            value_chain(
+                "",
+                [
+                    f"{label}={str(getattr(self, label, 'Not set'))}"
+                    for label in self._printable_attributes
+                ],
+                [f"{key}={str(self.get(key, 'Not set'))}" for key in self.keys()],
```

Comment:
Shouldn't need to instantiate lists here, also, if this derives from `dict`, `keys` is the default iterable
```suggestion
                (
                    f"{label}={str(getattr(self, label, 'Not set'))}"
                    for label in self._printable_attributes
                ),
                (f"{key}={str(self.get(key, 'Not set'))}" for key in self),
```